From 5838d235a395e221968e1f024bfaf8650278a522 Mon Sep 17 00:00:00 2001
From: Muchun Song <songmuchun@bytedance.com>
Date: Tue, 13 Jul 2021 02:18:56 +0000
Subject: [PATCH] arm64: mm: hugetlb: add support for free vmemmap pages of
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 5838d235a395e221968e1f024bfaf8650278a522

 HugeTLB

hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I3ZCW9
CVE: NA

-------------------------------------------------

The preparation of supporting freeing vmemmap associated with each
HugeTLB page is ready, so we can support this feature for arm64.

Signed-off-by: Muchun Song <songmuchun@bytedance.com>
Signed-off-by: Chen Huang <chenhuang5@huawei.com>
Signed-off-by: Nanyong Sun <sunnanyong@huawei.com>
Reviewed-by: Tong Tiangen <tongtiangen@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/Kconfig b/fs/Kconfig
index d31f23879f26..83c56571d0bc 100644
--- a/fs/Kconfig
+++ b/fs/Kconfig
@@ -238,7 +238,7 @@ config HUGETLB_PAGE
 
 config HUGETLB_PAGE_FREE_VMEMMAP
 	def_bool HUGETLB_PAGE
-	depends on X86_64
+	depends on X86_64 || ARM64
 	depends on SPARSEMEM_VMEMMAP
 
 config HUGETLB_PAGE_FREE_VMEMMAP_DEFAULT_ON
-- 
2.26.2

